src/config.js

```diff
@@ -1,9 +1,16 @@
 // @flow
 /* global $Shape */
-import type { Config } from './types';
+import type { AdvancedConfig, Config } from './types';
 import defaults from './defaults';
 
 export const applyDefaults = (config: $Shape<Config> = {}): Config => ({
   ...defaults,
   ...config
 });
+
+export const applyAdvancedDefaults = (
+  config: $Shape<AdvancedConfig> = {}
+): AdvancedConfig => ({
+  ...defaults,
+  ...config
+});
```

Comment:
I tried to make the `applyDefaults` function above generic, but was getting an `empty` return type from Flow, so I resorted to having distinct functions. Any better suggestions are welcome.